src/PointwiseFunctions/AnalyticSolutions/Poisson/Moustache.cpp

```diff
@@ -0,0 +1,116 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/Poisson/Moustache.hpp"
+
+#include <algorithm>
+#include <array>
+#include <pup.h>  // IWYU pragma: keep
+
+#include "DataStructures/DataVector.hpp"     // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/MakeWithValue.hpp"
+// IWYU pragma: no_forward_declare Tensor
+
+namespace Poisson {
+namespace Solutions {
+
+namespace {
+
+template <size_t Dim>
+tnsr::I<DataVector, Dim> auxiliary_field(
+    const tnsr::I<DataVector, Dim>& x,
+    const DataVector& precomputed_norm_square) noexcept;
+
+template <>
+tnsr::I<DataVector, 1> auxiliary_field<1>(
+    const tnsr::I<DataVector, 1>& x,
+    const DataVector& precomputed_norm_square) noexcept {
+  auto result = make_with_value<tnsr::I<DataVector, 1>>(x, 0.);
+  const auto& x_d = get<0>(x);
+  get<0>(result) = sqrt(precomputed_norm_square) *
+                   (0.25 - 3. * x_d + 7.5 * square(x_d) - 5. * pow(x_d, 3));
+  return result;
+}
+
+template <>
+tnsr::I<DataVector, 2> auxiliary_field<2>(
+    const tnsr::I<DataVector, 2>& x,
+    const DataVector& precomputed_norm_square) noexcept {
+  auto result = make_with_value<tnsr::I<DataVector, 2>>(x, 0.);
+  for (size_t d = 0; d < 2; d++) {
+    const auto& x_d = x.get(d);
+    const auto& x_p = x.get((d + 1) % 2);
+    result.get(d) =
+        sqrt(precomputed_norm_square) * x_p * (1. - x_p) *
+        (0.5 - 3.5 * x_d + 7.5 * square(x_d) - 5. * pow(x_d, 3) - x_p +
+         2. * x_d * x_p + square(x_p) - 2. * x_d * square(x_p));
+  }
+  return result;
+}
+
+}  // namespace
+
+template <size_t Dim>
+tuples::TaggedTuple<Field, AuxiliaryField<Dim>> Moustache<Dim>::field_variables(
+    const tnsr::I<DataVector, Dim>& x) const noexcept {
+  auto field = make_with_value<Scalar<DataVector>>(x, 1.);
+  for (size_t d = 0; d < Dim; d++) {
+    field.get() *= x.get(d) * (1. - x.get(d));
+  }
+  auto norm_square = make_with_value<DataVector>(get<0>(x), 0.);
+  for (size_t d = 0; d < Dim; d++) {
+    norm_square += square(x.get(d) - 0.5);
+  }
+  field.get() *= pow(norm_square, 3. / 2.);
+  return {std::move(field), auxiliary_field(x, norm_square)};
+}
+
+template <>
+tuples::TaggedTuple<::Tags::Source<Field>, ::Tags::Source<AuxiliaryField<1>>>
+Moustache<1>::source_variables(const tnsr::I<DataVector, 1>& x) const noexcept {
+  auto field_source = make_with_value<Scalar<DataVector>>(x, 1.);
+  const auto& x1 = get<0>(x);
+  field_source.get() *= (0.875 - 8.5 * x1 + 28.5 * square(x1) -
```

Comment:
I have no idea where these numbers are coming from and the documentation isn't helping. Could you clarify wherever is most appropriate?